config/management/src/smoke_test.rs

```diff
@@ -0,0 +1,167 @@
+        Ok((self.configs.clone(), self.faucet_key.clone()))
+    }
+}
+
+#[test]
+fn smoke_test() {
+    let helper = StorageHelper::new();
+    let num_validators = 5;
+    let shared = "_shared";
+    let association = "association";
+    let association_shared = association.to_string() + shared;
+
+    // Step 1) Prepare the layout
+    let mut layout = Layout::default();
+    layout.association = vec![association_shared.to_string()];
+    layout.operators = (0..num_validators)
+        .map(|v| (v.to_string() + shared))
+        .collect();
+
+    let mut common_storage = helper.storage(crate::constants::COMMON_NS.into());
+    let layout_value = Value::String(layout.to_toml().unwrap());
+    common_storage
+        .set(crate::constants::LAYOUT, layout_value)
+        .unwrap();
+
+    // Step 2) Set association key
+    helper.initialize(association.into());
+    helper
+        .association_key(&association, &association_shared)
+        .unwrap();
+
+    // Step 3) Prepare validators
+    let mut configs = Vec::new();
+    for i in 0..num_validators {
+        let storage = helper.storage(i.to_string());
+        let ns = i.to_string();
+        let ns_shared = ns.clone() + shared;
+        helper.initialize(ns.clone());
+
+        let mut config = NodeConfig::default();
+
+        let mut network = NetworkConfig::default();
+        network.discovery_method = DiscoveryMethod::Onchain;
+        config.validator_network = Some(network);
+
+        let mut network = NetworkConfig::default();
+        network.discovery_method = DiscoveryMethod::Onchain;
+        config.full_node_networks = vec![NetworkConfig::default()];
+        config.randomize_ports();
+
+        let validator_network = config.validator_network.as_mut().unwrap();
+        let validator_network_address = validator_network.listen_address.clone();
+        let identity_key = storage
+            .export_private_key(libra_global_constants::VALIDATOR_NETWORK_KEY)
+            .unwrap();
+        let identity_key: x25519::PrivateKey = identity_key.to_bytes().as_ref().try_into().unwrap();
+        // This is a deprecated field
+        let signing_key = Ed25519PrivateKey::generate_for_testing();
+        validator_network.network_keypairs = Some(NetworkKeyPairs::load(identity_key, signing_key));
+
+        let fullnode_network = &mut config.full_node_networks[0];
+        let identity_key = storage
+            .export_private_key(libra_global_constants::FULLNODE_NETWORK_KEY)
+            .unwrap();
+        let identity_key = identity_key.to_bytes().as_ref().try_into().unwrap();
+        // This is a deprecated field
+        let signing_key = Ed25519PrivateKey::generate_for_testing();
+        fullnode_network.network_keypairs = Some(NetworkKeyPairs::load(identity_key, signing_key));
+        let fullnode_network_address = fullnode_network.listen_address.clone();
+
+        configs.push(config);
+
+        helper.operator_key(&ns, &ns_shared).unwrap();
```

Comment:
those helper functions are so hard to read, spend a few mins realizing this is to fetch from local and submit to remote

Reply:
ahh, the intent is to ensure the command-line works as possible, I guess I could add comments to storage_helper... thoughts on what the confusion stemmed from?

Reply:
I'd expect something like `submit_[operator_key|validator_config]`, `operator_key` feels the function only returns the operator key without doing anything else